src/core/src/energy/functions.rs

```diff
@@ -299,6 +299,34 @@ impl PairPotential for Buckingham {
     }
 }
 
+struct MorsePotential {
```

Comment:
The struct is missing documentation. You can take inspiration from the other potentials in this file.